src/Symfony/Component/Validator/Constraints/TwigValidator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Twig\Environment;
+use Twig\Error\Error;
+use Twig\Loader\ArrayLoader;
+use Twig\Source;
+
+/**
+ * @author Mokhtar Tlili <tlili.mokhtar@gmail.com>
+ */
+class TwigValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Twig) {
+            throw new UnexpectedTypeException($constraint, Twig::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!\is_scalar($value) && !$value instanceof \Stringable) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = (string) $value;
+
+        $prevErrorHandler = set_error_handler(static function ($level, $message, $file, $line) use (&$prevErrorHandler) {
+            if (\E_USER_DEPRECATED === $level) {
+                $templateLine = 0;
+                if (preg_match('/ at line (\d+)[ .]/', $message, $matches)) {
+                    $templateLine = $matches[1];
+                }
+
+                throw new Error($message, $templateLine);
+            }
+
+            return $prevErrorHandler ? $prevErrorHandler($level, $message, $file, $line) : false;
+        });
+
+        try {
+            $twig = new Environment(new ArrayLoader([$value]));
```

Comment:
This is bad. It has to be possible to inject a configured Twig environment. Otherwise, I can only validate if a given template would work in a Twig environment without any extensions which is kind of useless in most projects because you usually add own functions, filters etc. to Twig.

Reply:
Thanks for your valuable comment. Absolutely make sense